entity-type-creator/components/EntityTypeFieldEditor.tsx

```diff
@@ -165,9 +165,7 @@ export default function EntityTypeFieldEditor({
               labelId={`${parentName}-${field.name}-data-type`}
               fullWidth
               value={field.dataType?.dataType}
-              onChange={(e) =>
-                onChange({ ...field, dataType: { ...field.dataType, dataType: e.target.value as DataTypeValue } })
-              }
+              onChange={(e) => onChange({ ...field, dataType: { dataType: e.target.value as DataTypeValue } })}
```

Comment:
throws out the sub-properties of `dataType` on change, to prevent keeping `itemDataType` or `properties` when it no longer makes sense